src/services/getGlobalHooksForRouter.ts

```diff
@@ -0,0 +1,19 @@
+import { AfterRouteHookLifecycle, BeforeRouteHookLifecycle, Hooks } from '@/types/hooks'
+import { RouterOptions } from '@/types/router'
+import { RouterPlugin } from '@/types/routerPlugin'
+import { AsArray } from '@/types/utilities'
+
+export function getGlobalHooksForRouter(options: RouterOptions = {}, plugins: RouterPlugin[] = []): Hooks {
+  return {
+    onBeforeRouteEnter: getHooksForLifecycle('onBeforeRouteEnter', options, plugins),
+    onBeforeRouteUpdate: getHooksForLifecycle('onBeforeRouteUpdate', options, plugins),
+    onBeforeRouteLeave: getHooksForLifecycle('onBeforeRouteLeave', options, plugins),
+    onAfterRouteEnter: getHooksForLifecycle('onAfterRouteEnter', options, plugins),
+    onAfterRouteUpdate: getHooksForLifecycle('onAfterRouteUpdate', options, plugins),
+    onAfterRouteLeave: getHooksForLifecycle('onAfterRouteLeave', options, plugins),
+  }
+}
+
+function getHooksForLifecycle<T extends BeforeRouteHookLifecycle | AfterRouteHookLifecycle>(lifecycle: T, options: RouterOptions, plugins: RouterPlugin[]): AsArray<Hooks[T]> {
+  return [options[lifecycle], ...plugins.map((plugin) => plugin[lifecycle])].flat().filter((hook) => hook !== undefined) as AsArray<Hooks[T]>
```

Comment:
eww dude. lol

```suggestion
  const optionsHooks = options[lifecycle] ?? []
  const pluginHooks = plugins.map((plugin) => plugin[lifecycle])
  const allHooks = [optionsHooks, ...pluginHooks]
    .flat()
    .filter((hook) => hook !== undefined)

  return allHooks as AsArray<Hooks[T]>
```

Reply:
I think yours is eww lol so many consts 😂 How about this
```suggestion
  return [
    options[lifecycle],
    ...plugins.map((plugin) => plugin[lifecycle]),
  ].flat().filter((hook) => hook !== undefined) as AsArray<Hooks[T]>
```

Reply:
Also a small private utility, I don't care to much about this especially since there's already an `as` 😂

Reply:
Or maybe this
```suggestion
  const hooks = [
    options[lifecycle],
    ...plugins.map((plugin) => plugin[lifecycle]),
  ].flat().filter((hook) => hook !== undefined)

  return hooks as AsArray<Hooks[T]>
```